File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.go
func searchMatrix(matrix [][]int, target int) bool {
    n:=len(matrix)
    m:=len(matrix[0])

    low:=0
    high:=n*m-1

    for low<=high{
        guess:=(low+high)/2

        row:=guess/m
        col:=guess%m

        if matrix[row][col]==target{
            return true
        }

        if matrix[row][col]<target{
            low=guess+1
        }else{
            high=guess-1
        }
    }
    return false
}

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let n=matrix.length;
    let m=matrix[0].length;

    let low=0;
    let high=n*m-1;

    while(low<=high){
        let guess=Math.floor((low+high)/2);

        let row=Math.floor(guess/m);
        let col=guess%m;

        if(matrix[row][col]==target){
            return true;
        }
        if(matrix[row][col]<target){
            low=guess+1;
        }
        else{
            high=guess-1;
        }
    }
    return false;
};